### ISSUE TITLE
Dの冒険Ⅶ: Hang after investigating the grave

### ISSUE THREAD

**Issue by Ghabry** [EasyRPG/Player#2106]
#### Name of the game:
Dの冒険Ⅶ

https://viprpgkohaku2016.wixsite.com/kohaku2016/no-20
[Direct DL link](https://www.mediafire.com/file/e6i4cmms42wyry5/vipkh16_20_%25EF%25BC%25A4%25E3%2581%25AE%25E5%2586%2592%25E9%2599%25BA7%25E3%2580%2580%25EF%25BD%259ED%25E3%2581%25AE%25E8%25AC%258E%25EF%25BC%2581%25EF%25BC%259F%25EF%25BD%259E.zip/file)

#### Attach files (as a .zip archive or link them)
[Save07.zip](https://github.com/EasyRPG/Player/files/4432686/Save07.zip)

#### Describe the issue in detail and how to reproduce it:

Investigate the grave directly above you. It fades to a scene where it says "Chapter 8" and then it hangs with "Event 2 reached execution limit"

Good news: Not a regression. Hangs in 0.5.4 and newer

**Comment by Ghabry** [comment]
Same as #1733 : Needs proper MIDI Ticks support

**Comment by Ghabry** [comment]
This game only uses a ">=" measurement to decide if enough time has past. So implementing a not precise version of midi ticks would be enough to fix it. I was already able to implement a sloppy version of GetTicks for fmmidi. In my test it has around the same value as harmony, with an error of +-100. Will also try to get it working for Wildmidi.

There is another more interesting bug here: I'm lucky that I used SdlAudio instead of SdlMixerAudio otherwise wouldn't have noticed it:

The game quickly does 2 "BgmPlay" event calls after each other. Somehow the 2nd BgmPlay event is lost. When BgmStop is called we synchronize on a mutex with SDL_Lock/UnlockAudio and wait until it is safe to delete the bgm decoder.
It seems SDL_Lock/Unlock doesn't behave really like a mutex (maybe allows multiple locks?): It doesn't wait on the Lock, the Bgm is not destroyed and BgmPlay fails because there is no free channel.

tl;dr: std::mutex fixes this.

**Comment by Ghabry** [comment]
The observation was incorrect, SDL_Lock/UnlockAudio works as expected. Was another bug in the code.
